### PR TITLE
chore: add docs for tera features for the template documenation

### LINKED ISSUE #2048: template documentation improvements
I would like to have a section in the docs that covers the basics of tera in mise, this I think may repeat a lot of what is already in the tera docs, but having it in one place would help users learn tera while also learning the functions that mise builds on top of tera.

### PR DESCRIPTION
**Pull request by erickguan**
Closes #2048

Please raises comments for the document structure and general documentation improvements.

This should be merged after closing #2049.

### CODE REVIEW

**Comment by erickguan** [comment]
Fixed!

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -134,9 +134,13 @@ export default defineConfig({
     },
   },
   markdown: {
-    // languages: [
-    //   "elisp"
-    // ]
+    config(md) {
```
Comment (jdx) [inline docs/.vitepress/config.ts]: what is this for?

File: docs/templates.md [hunk 2]
```diff
@@ -1,95 +1,504 @@
...
+
+Mise offers additional tests:
+
+- `if path is dir` – Checks if the provided path is a directory.
+- `if path is file` – Checks if the path points to a file.
+- `if path is exists` – Checks if the path exists.
+- `if version is semver_matching(String)` – Confirms if a version string matches
+  a semantic version pattern (e.g., `if "1.10.2" is semver_matching("^1.10.0")`).
+
+### Example Templates
```
Comment (jdx) [inline docs/templates.md:284]: all of these examples look wrong in several places, have you actually tested them?

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (erickguan) [reply docs/.vitepress/config.ts]: `vitepress` seems doing some magic on inline code block. e.g., vitepress doesn't render `{{`. `[`vitepress`](https://github.com/vuejs/vitepress/discussions/480)` recommends to either use `<code v-pre>{{ 1 + 1 }}</code>` or this config.

File: docs/templates.md (hunk 2 above)
Comment (erickguan) [reply docs/templates.md:284]: Ugh, my ChatGPT practice is not going so hot. I configured a few templates and build some examples. The examples should work now.

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (erickguan) [reply docs/.vitepress/config.ts]: Fixed with HTML blocks.

**Pull request merged**